File: Selection Sort/selection_sort.py
def selection_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        min_index = i
        # Поиск минимального в подмассиве arr[i..n-1]
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j
        # Меняем местами текущий элемент и найденный минимум
        arr[i], arr[min_index] = arr[min_index], arr[i]

# Пример использования
numbers = [64, 25, 12, 22, 11]
selection_sort(numbers)
print(numbers)  # Результат: [11, 12, 22, 25, 64]

# Здесь min_index – это позиция, на которой находится минимальный элемент среди всех, что ещё не отсортированы. 
# После нахождения мы меняем местами элемент arr[i] (начало неотсортированной области) и элемент arr[min_index].

File: Selection Sort/selection_sort.js
function selectionSort(arr) {
  const n = arr.length;
  
  for (let i = 0; i < n - 1; i++) {
    let minIndex = i;
    
    // Находим индекс минимального элемента
    for (let j = i + 1; j < n; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }
    
    // Меняем местами минимальный элемент с текущим
    if (minIndex !== i) {
      [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]];
    }
  }
  
  return arr;
}

// Пример использования
console.log(selectionSort([64, 34, 25, 12, 22, 11, 90]));
// Результат: [11, 12, 22, 25, 34, 64, 90]
